drivers/gpio/gpio_efinix_sapphire.c

```diff
@@ -0,0 +1,229 @@
+/*
```

Comment:
Please ensure that this (and all of the drivers you have added in this change set) are added to `tests/drivers/build_all`

Reply:
1. I've included a GPIO test case in this pull request. Could you please review it and let me know if it fulfils your requirements?

2. There isn't a serial directory in tests/drivers/build_all. Would you like me to create one?